content/docs/tutorials/get-started/data-pipelines.md

```diff
@@ -137,59 +137,167 @@ stage as <abbr>dependencies</abbr> of another one, we can describe a sequence of
+with the same set of options:
 
-$ dvc run -f train.dvc \
+```dvc
+$ dvc run -n train \
+          -p train.seed,train.n_estimators \
           -d src/train.py -d data/features \
           -o model.pkl \
           python src/train.py data/features model.pkl
 ```
 
-This would be a good point to commit the changes with Git. This includes any
-`.gitignore` files, and `dvc.yaml` — which describes our pipeline.
+Please, check the `dvc.yaml` again, it should have plus one stage now.
+
+</details>
 
-> 📖 See also the `dvc pipeline` command.
+This should be a good point to commit the changes with Git. This include
+`.gitignore`, `dvc.lock`, and `dvc.yaml` — which describes our pipeline.
 
 ## Reproduce
 
-Imagine you're just cloning the <abbr>repository</abbr> created so far, in
-another computer. It's extremely easy for anyone to reproduce the result
-end-to-end, by using `dvc repro`.
+The whole point of creating this `dvc.yaml` pipeline file is an ability to
+reproduce it:
+
+```dvc
+$ dvc repro
+```
 
 <details>
 
-### 👉 Expand to simulate a fresh clone of this repo
+### ⚙️ Expand to make some fun with it
 
-Move to another location in your file system and do this:
+Let's try to play a little bit with it. First, let's try to change one of the
+parameters for the training stage:
 
 ```dvc
-$ git clone https://github.com/iterative/example-get-started
-$ cd example-get-started
-$ git checkout 7-train
+$ vim params.yaml
 ```
 
-</details>
+Change `n_estimators` to `100` and run `dvc repro`, you should see:
 
 ```dvc
-$ dvc repro train.dvc
+$ dvc repro
+Stage 'prepare' didn't change, skipping
+Stage 'featurize' didn't change, skipping
+Running stage 'train' with command: ...
 ```
 
-`train.dvc` is used because it's the last stage file so far; It describes what
-code and data to use to regenerate a final result (ML model). For stages that
-<abbr>output</abbr> any of its <abbr>dependencies</abbr>, we can in turn get the
-same info, and so on.
+It's detected that only training should be run and skipped everything else! All
+the intermediate results are being reused.
+
+Now, let's change it back to `50` and run `dvc repro` again:
+
+```dvc
+$ dvc repro
+Stage 'prepare' didn't change, skipping
+Stage 'featurize' didn't change, skipping
+Restored stage 'train' from run-cache
+```
+
+Same as before, no need to run `prepare`, `featurize`, etc ... but, it doesn't
+run even `train` again this time! It cached the previous run with the same set
+of inputs (parameters + data) and reused it now!
+
+</details>
+
+<details>
+
+### 💡Expand to see what happens under the hood
+
+`dvc.repro` relies on the DAG definition that it reads from the `dvc.yaml` file
+and also on the `dvc.lock` file to determine what exactly needs to be run.
+
+`dvc.lock` file is similar to `.dvc` files and captures hashes (in most cases
+`md5`s) of the dependencies, values of the parameters that were used, it can be
+considered a _state_ of the pipeline:
```

Comment:
```suggestion
`dvc.lock` file is similar to `.dvc` files and captures file hashes (
`md5` fields) of the dependencies, values of the parameters that were used. It can be
considered a _state_ of the pipeline:
```

Reply:
Not sure about term "state" here since it overlaps with `dvc status`.

Reply:
can discuss this next time :)